File: 346.py
ans = 1
n =1000000000000
m = {}
for i in range(2,n):
    t = i*i
    j = 1+i
    while j+t<n:
        j+=t
        t*=i
        if m[j]==1:
            continue
        ans+=j
        m[j]=1
        print(j)
print(ans)

File: 346.cpp
#include<bits/stdc++.h>
#define ull unsigned long long
using namespace std;
vector<ull> v;
int main()
{
    auto n = 1e12;
    ull ans=1;
    for(ull i=2;i*i<n;i++)
    {
        auto t = i*i;
        auto j = i+1;
        while(j+t<n)
        {
            j+=t;
            t*=i;
            v.push_back(j);
        }
    }
    sort(v.begin(),v.end());
    auto last = unique(v.begin(),v.end());
    v.erase(last,v.end());

    for(auto i : v)
        ans+=i;
    cout<<ans;
}
